WireUI/Sources/WireSidebar/SidebarView.swift

```diff
@@ -0,0 +1,276 @@
+                conversationFilter.label(iconSize, isActive: self.conversationFilter == conversationFilter) {
+                    self.conversationFilter = conversationFilter
+                }
+            }
+
+            Text("sidebar.contacts.title", bundle: .module)
+                .wireTextStyle(.h2)
+                .padding(.horizontal, 8)
+                .padding(.vertical, 12)
+                .padding(.top, 12)
+            SidebarMenuItem(
+                icon: "person.badge.plus",
+                iconSize: iconSize
+            ) {
+                Text("sidebar.contacts.connect.title", bundle: .module)
+            } action: {
+                connectAction()
+            }
+        }
+        .padding(.horizontal, 16)
+    }
+}
+
+// MARK: - SidebarData.ConversationFilter + label
+
+private extension SidebarConversationFilter? {
+
+    @MainActor
+    func label(_ iconSize: CGSize?, isActive: Bool, action: @escaping () -> Void) -> SidebarMenuItem {
+        let text: Text
+        let icon: String
+
+        switch self {
+        case .none:
+            text = Text("sidebar.conversation_filter.all.title", bundle: .module)
+            icon = "text.bubble"
+
+        case .favorites:
+            text = Text("sidebar.conversation_filter.favorites.title", bundle: .module)
+            icon = "star"
+
+        case .groups:
+            text = Text("sidebar.conversation_filter.groups.title", bundle: .module)
+            icon = "person.3"
+
+        case .oneOnOne:
+            text = Text("sidebar.conversation_filter.oneOnOneConversations.title", bundle: .module)
+            icon = "person"
+
+        case .archived:
+            text = Text("sidebar.conversation_filter.archived.title", bundle: .module)
+            icon = "archivebox"
+        }
+
+        return SidebarMenuItem(
+            icon: icon,
+            iconSize: iconSize,
+            isLink: false,
+            isHighlighted: isActive,
+            title: { text },
+            action: action
+        )
+    }
+}
+
+// MARK: - View Modifiers + Environment
+
+extension View {
+    func sidebarBackgroundColor(_ sidebarBackgroundColor: Color) -> some View {
+        modifier(SidebarBackgroundColorViewModifier(sidebarBackgroundColor: sidebarBackgroundColor))
+    }
+}
+
+private extension EnvironmentValues {
+    var sidebarBackgroundColor: Color {
+        get { self[SidebarBackgroundColorKey.self] }
+        set { self[SidebarBackgroundColorKey.self] = newValue }
+    }
+}
+
+struct SidebarBackgroundColorViewModifier: ViewModifier {
+    var sidebarBackgroundColor: Color
+    func body(content: Content) -> some View {
+        content
+            .environment(\.sidebarBackgroundColor, sidebarBackgroundColor)
+    }
+}
+
+private struct SidebarBackgroundColorKey: EnvironmentKey {
+    static let defaultValue = Color(uiColor: .systemGray5)
+}
+
+// MARK: - Previews
```

Comment:
Consider putting preview code within `#if DEBUG`. I think it is a good habit because:

- We ship less code
- In cases where not everything is private it prevents us from accidentally depending on preview code

Reply:
Agreed offline that it's ok.